actors/evm/src/interpreter/precompiles.rs

```diff
@@ -71,24 +71,36 @@ fn read_u256_infalliable(buf: &[u8], start: usize) -> U256 {
 }
 
 fn ec_recover<RT: Primitives>(rt: &RT, input: &[u8]) -> PrecompileResult {
-    let mut buf = [0u8; 128];
-    buf[..input.len().min(128)].copy_from_slice(&input[..input.len().min(128)]);
-
+    if input.len() != 128 {
+        return Err(PrecompileError::IncorrectInputSize);
+    }
     let mut hash = [0u8; SECP_SIG_MESSAGE_HASH_SIZE];
     let mut sig = [0u8; SECP_SIG_LEN];
 
-    hash.copy_from_slice(&input[..32]);
-    sig.copy_from_slice(&input[64..]); // TODO this assumes input is exactly 65 bytes which would panic if incorrect
+    hash.copy_from_slice(&input[0..32]);
```

Comment:
FYI, we can avoid a copy here with https://doc.rust-lang.org/std/primitive.slice.html#impl-TryFrom%3C%26%27a%20%5BT%5D%3E.